### ISSUE TITLE
Camera management

### ISSUE THREAD

**Issue by GrilloPackerZ** [juliansteenbakker/mobile_scanner#897]
I would like to have better management of the cameras available on mobile devices. For example, on Samsungs (S21, S22 and S23) the rear camera that opens when starting is the ultra wide camera, whose focal length has several problems when it comes to reading a barcode. And I can't use the main camera because I don't have that control, since the only change we can make is choose whether to use the front and rear camera. At least as far as I could find.

**Comment by navaronbracke** [comment]
I do agree that we should allow to select a specific camera for a given camera direction.

We had a similar bug report about focus issues on the web relating to the lack of this control as well.

**Comment by mwoelk** [comment]
**For web the solution could be quite simple:**

First: request permission to access the camera
```js
navigator.mediaDevices.getUserMedia({
   video: true,
});
```

Second (after permission is granted - otherwise will only show one device):
```js
navigator.mediaDevices.enumerateDevices()
```

This gives a list of all available devices incl. labels and deviceId.

Third use the device specified from the list by id:
```js
navigator.mediaDevices.getUserMedia({
   video: { deviceId: { exact: <deviceId> } },
});
```

I guess a nice way to have this work on cross-plattform is to create a Camera object containing information about the camera such as name, id and info if about facing direction (if available). A list of these camera objects should then be queryable via a `getCameras()` in `MobileScannerController`. Next, the `switchCamera` method could get an optional parameter `camera:` that would accept one of the previously retrieved cameras.

From a previous project I also remember that when getting a list of cameras on android, there might be cameras listed that require system permission. those should be filtered from the list because it may lead to permission errors if a user wants to select them.

Please give me some feedback on the proposed solution and let me get started drafting a PR for this.

**Comment by navaronbracke** [comment]
@mwoelk Thanks for the detailed idea for a solution to this problem.

I agree that your approach can fix this problem. However, I want to land the platform interface rework first, before we add this change, so that it can be added easily for each platform (we could even ship support for this feature in stages)

**Comment by mwoelk** [comment]
Hey @navaronbracke,

Sounds great. Let me know if I can support. Got some time and a project where this feature would really help.

**Comment by navaronbracke** [comment]
If you could test https://github.com/juliansteenbakker/mobile_scanner/pull/916 that would be nice. I plan to put version 5.0.0 in beta for a bit, until we know that it doesn't contain any regressions, _and_ it supports all features that we used to support.

Mainly `JSQR` is not included anymore (this is probably intended, as ZXing supports more features?), and we should implement the remaining methods from the platform interface.

**Comment by oznecniV97** [comment]
Are there any update/opened PR on this?
How actually the package choose the camera? First in a list or something similar?

Using web version on my Galaxy S10 the default one is the 0.5x and somethimes the scan does not work well.

**Comment by anubyssl** [comment]
Agreed that having some control over which camera gets opened would be a very nice improvement, even if it is up to the user of this library to pass in the exact camera name/ID.

It seems that the telephoto or wide angle camera gets opened by default on many devices, which results in some barcodes struggling to get detected - especially barcodes with finer detail like PDF417.

**Comment by EbenauDev** [comment]
Any updates on this PR?

For my use case, I need a way to select a specific camera on web and pass it to MobileScanner, or some way to handle multiple cameras in the web implementation.

Do you have any suggestions or workarounds?

**Comment by navaronbracke** [comment]
I know there were improvements to web, see https://github.com/juliansteenbakker/mobile_scanner/blob/develop/lib/src/web/mobile_scanner_web.dart

But I think we would need a web only API to query the available cameras (you could do this with package:web, we could provide a shim) and pass in an optional camera id or something

**Comment by EbenauDev** [comment]
In my project, I use the `package :camera` and call `availableCameras() ``to list the available cameras. Then, I use a dropdown button to let the user select the camera they want.

I guess it might be possible to change this behavior or create a workaround in this line of code?

<img width="863" height="428" alt="Image" src="https://github.com/user-attachments/assets/51069b7d-2ebd-44a5-821f-b60b92a0682f" />

**Comment by navaronbracke** [comment]
Yes, I was thinking of something like `availableCameras()`, which then you could pass to start().

For example "String cameraId" or something

cc @juliansteenbakker what was the original idea behind saving the device id in localstorage again?
